permissions/api_permissions/graphene_permissions.py

```diff
@@ -91,6 +92,46 @@ def has_permission(cls, info: ResolveInfo) -> bool:
         return info.context.user.is_authenticated
 
 
+class ApplicationEventScheduleResultPermission(BasePermission):
+    @classmethod
+    def has_mutation_permission(cls, root: Any, info: ResolveInfo, input: dict) -> bool:
+        if not cls.has_permission(info):
+            return False
+
+        pk = input.get("application_event_schedule")
+        if not pk:
+            return False
+
+        schedule = ApplicationEventSchedule.objects.get(pk=pk)
+        application = schedule.application_event.application
+        service_sector = application.application_round.service_sector
+
+        return can_manage_service_sectors_applications(
+            info.context.user, service_sector
+        )
+
+    @classmethod
+    def has_permission(cls, info: ResolveInfo) -> bool:
+        user = info.context.user
+
+        if user and user.is_anonymous:
+            return False
+
+        service_sector_roles = user.service_sector_roles.all()
```

Comment:
Is there a need to check if `user` is `None`?
Just thinking if it is possible for this code to crash because `None` type does not have `service_sector_roles` property?

Reply:
good point, I guess we can make this `or`

Reply:
OR not! :smile: I'll refactor anyhow